iModelJsNodeAddon/presentation/ECPresentationSerializer.cpp

```diff
@@ -1322,3 +1322,158 @@ RulesetVariables IModelJsECPresentationSerializer::GetRulesetVariablesFromJson(B
+        if (rc.IsValid())
+            path.push_back(rc);
+        return false;
+        });
+
+    return path;
+    }
+
+#define PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_FilterExpression                   "expression"
+#define PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_SelectClassName                    "selectClassName"
+#define PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances                   "relatedInstances"
+#define PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_PathToProperty    "pathFromSelectToPropertyClass"
+#define PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_IsRequired        "isRequired"
+#define PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_Alias             "alias"
+#define PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_RelationshipAlias "relationshipAlias"
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+static bvector<RelatedClassPath> ParseInstanceFilterRelatedInstances(IConnectionCR connection, BeJsConst json)
+    {
+    if (!json.isArray())
+        DIAGNOSTICS_HANDLE_FAILURE(DiagnosticsCategory::Serialization, "Expected instance filter JSON member `" PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances "` to be an array");
+
+    bvector<RelatedClassPath> instances;
+    json.ForEachArrayMember([&](BeJsConst::ArrayIndex, BeJsConst instanceDef)
+        {
+        if (!instanceDef.isObject() || !instanceDef.hasMember(PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_PathToProperty))
+            return false;
+
+        RelatedClassPath pathToProperty = IModelJsECPresentationSerializer::GetRelatedClassPathFromJson(connection.GetECDb(), instanceDef[PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_PathToProperty], true);
+        if (pathToProperty.empty())
+            return false;
+
+        bool hasAlias = false;
+        if (instanceDef.hasMember(PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_Alias) && instanceDef[PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_Alias].isString())
+            {
+            Utf8String alias = instanceDef[PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_Alias].ToUtf8CP();
+            pathToProperty.back().GetTargetClass().SetAlias(alias);
+            hasAlias = true;
+            }
+
+        if (instanceDef.hasMember(PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_RelationshipAlias) && instanceDef[PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_RelationshipAlias].isString())
+            {
+            Utf8String alias = instanceDef[PRESENTATION_JSON_ATTRIBUTE_InstanceFilterDefinition_RelatedInstances_RelationshipAlias].ToUtf8CP();

```

Comment:
I think this `if` missing `hasAlias = true` statement. I seed that it was missing it before but I think `hasAlias` flag is here to determine if atleast one of the aliases is present.

Reply:
Agree, made a change to set `hasAlias = true` when there is a relationship alias